### ISSUE TITLE
VSCode freezing when adding folder to workspace while working on remote Linux machine

### ISSUE THREAD

**Issue by ravitejapavuluri** [microsoft/vscode-remote-release#10168]
Type: <b>Performance Issue</b>

I have SSH host configured 

1. When I click at the bottom left side connect to remote host and I can connect to the selected host.
2. Once connected to the remote host, click add folder to workspace, it freezes and doesn't quit. I need to end task task from task manager.
3. I sometimes ssh to the same machine with different users and work on the same folder.

VS Code version: Code 1.92.1 (eaa41d57266683296de7d118f574d0c2652e1fc4, 2024-08-07T20:16:39.455Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Connection to 'SSH: ycvghfd500_int1' could not be established  Canceled

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 2995)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.71GB (14.19GB free)|
|Process Argv|--crash-reporter-id 21c459a2-a353-4566-adaf-1f7dd75c49d7|
|Screen Reader|no|
|VM|0%|

Connection to 'SSH: ycvghfd500_int1' could not be established  Canceled
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	   148	 34084	code main
    0	   204	  2168	window [3] (Welcome - Untitled (Workspace) - Visual Studio Code)
    0	    84	 11636	   utility-network-service
    0	    96	 17192	fileWatcher [3]
    0	   100	 19160	ptyHost
    0	   104	 20560	window [4] (Issue Reporter)
    0	    81	 31048	   crashpad-handler
    0	   242	 34196	   gpu-process
    0	   133	 51600	shared-process
Connection to 'SSH: pjclx101_int1' could not be established  Canceled
```

</details>
<details>
<summary>Workspace Info</summary>

```

Connection to 'SSH: ycvghfd500_int1' could not be established  Canceled;
```

</details>
Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-c:31111717
wkspc-ranged-t:31111713

```

</details>

**Comment by joshspicer** [comment]
Thank you for taking the time to open an issue.  This sounds to be related to the Remote - SSH extension, correct?  Could you please also follow the troubleshooting steps [outlined here](https://github.com/microsoft/vscode-remote-release/wiki/Remote-SSH-troubleshooting) and report back?

I'm a bit confused by how the error message you shared (`could not be established Canceled`) is related? Could you please share the entire log where you see this message?

**Comment by TedThemistokleous** [comment]
I ran into this similar issue too after last week's update. Verbatim had to force close windows that were freezing and broken. I just reinstalled using the latest version, issue seems to be messing. Likely something with your update package is broken.

**Comment by ravitejapavuluri** [comment]
Yes, this is from the latest update. My friend is also facing same issue. After connection to remote SSH, he cannot add folder to workspace. I cannot take the logs, it just freezes and I need to kill from taskmanager.

Could be from the SSH extension but not sure.

**Comment by jbiams77** [comment]
This is also happening to me and my team. Since updating to 1.92.2, when I have vscode connected over Remote-SSH extension, when I add a new folder to workspace, my vscode freezes.

**Comment by watarungurunnn** [comment]
Same here.

**Comment by TedThemistokleous** [comment]
exact same behavior ^ and able to repo this. If it keeps occurring I seem to fix it by just doing another fresh reinstall. 

Saved session seems to get around this entirely but still a pain if say, server goes down and have to switch boxes.

**Comment by joshspicer** [comment]
Related: https://github.com/microsoft/vscode/issues/225487

**Comment by zhaodiaoer** [comment]
I ran into this issue too, after click "reload anyway" button, the window will not respond to any action, it can only be restored by forcibly exiting and then reopen window. Interestingly, the newly opened window is in line with expectations.

**Comment by zhaodiaoer** [comment]
some new discoveries: when i open a new window -> connect remote ssh host -> click "Save Workspace As ..." ,  the window will not respond to any action until connection reconnected, although there are no opened code project.

**Comment by iwangbowen** [comment]
same here

**Comment by punmechanic** [comment]
I've been experiencing this for a few months now. Adding a folder to a workspace - or removing it - appears to visually do nothing. Usually, if I restart VSCode, I get two windows: One with the old previous state and one with the new workspace with the folder I wanted to add in.

This occurs over SSH to a darwin machine from Linux.

**Comment by sovietspaceship** [comment]
This also happens if you open a workspace in a remote ssh session. Possibly the most frustrating thing I've ever experienced in vscode. It works for a bit then completely hangs, even after reloading the window. The remote server just stops responding and must be killed

**Comment by josh-gleason** [comment]
I'm experiencing the same issue. If I wait a while after adding a folder to an empty workspace, then I eventually get a failure dialog and VSCode reconnects to the host. This occurs consistently if I add a folder to an empty workspace. Adding folders to a non-empty workspace seems to be fine.

Client: Windows 11 Pro
Server: Ubuntu 20.04
VSCode: 1.97-insider (same occurs on 1.96.2)

https://github.com/user-attachments/assets/a9697e42-94b0-45d3-b9e0-3c9131cf4c57

**Comment by andySquirro** [comment]
Just chiming in that this is still an issue. After adding another folder nothing happens until a "reload window" prompt shows up. After reloading the folder is correctly added (though adding another folder and manually reloading the window doesn't do anything - I've got to wait for the vscode prompt).

**Comment by zhaodiaoer** [comment]
Hi guys, are there any updates about this issue?

**Comment by bennm37** [comment]
I'm still having this issue on Mac, Code 1.92.2 - any updates?

**Comment by nanbiz** [comment]
Come on guys this is ridiculous we are in a team of 50 people and everyone has this problem almost daily, how is there so little energy invested in a tool that 75% of developers use? 
"Instead of fixing these basic features, just cram more AI into this, then people will be happier!!"
